File: Shirahama/module_oriented/member_module.py
# member module

members = {
    0:{'mail_address':"tpham@netprotections.co.jp", 'phone_number':"01-1234-5678", 'name_kanji':"Pham Thuc Hung", 'name_kana':"ファン　トュック　フン", 'sex_division':"男性", 'birth_day':"1991/09/16", 'address':"東京都中央区銀座1-10-6 銀座ファーストビル4階"},
    1:{'mail_address':"vle@netprotections.co.jp", 'phone_number':"01-1234-5678", 'name_kanji':"Le van Hiep", 'name_kana':"ファン　トュック　フン", 'sex_division':"男性", 'birth_day':"1991/09/16", 'address':"東京都中央区銀座1-10-6 銀座ファーストビル4階"},
    2:{'mail_address':"ksato@netprotections.co.jp", 'phone_number':"01-1234-5678", 'name_kanji':"佐藤 健太", 'name_kana':"サトウ　ケンタ", 'sex_division':"男性", 'birth_day':"1991/09/16", 'address':"東京都中央区銀座1-10-6 銀座ファーストビル4階"},
    3:{'mail_address':"snozaki@netprotections.co.jp", 'phone_number':"01-1234-5678", 'name_kanji':"野崎 才門", 'name_kana':"ノザキ　サイモン", 'sex_division':"男性", 'birth_day':"1991/09/16", 'address':"東京都中央区銀座1-10-6 銀座ファーストビル4階"},
    4:{'mail_address':"yiwata@netprotections.co.jp", 'phone_number':"01-1234-5678", 'name_kanji':"岩田 結", 'name_kana':"イワタ　ユイ", 'sex_division':"女性", 'birth_day':"1991/09/16", 'address':"東京都中央区銀座1-10-6 銀座ファーストビル4階"},
    5:{'mail_address':"tnagasaki@netprotections.co.jp", 'phone_number':"01-1234-5678", 'name_kanji':"長崎 太志", 'name_kana':"ナガザキ　タイシ", 'sex_division':"男性", 'birth_day':"1991/09/16", 'address':"東京都中央区銀座1-10-6 銀座ファーストビル4階"},
    6:{'mail_address':"kfukakawa@netprotections.co.jp", 'phone_number':"01-1234-5678", 'name_kanji':"深川 健太", 'name_kana':"フカケン", 'sex_division':"男性", 'birth_day':"1991/09/16", 'address':"東京都中央区銀座1-10-6 銀座ファーストビル4階"}
    }

# １．会員登録
def register():
    global members
    global member
    member={}
    member["mail_address"] = input("メールアドレスを入力してください：")
    member["phone_number"] = input("電話番号を入力してください(例：000-0000-0000)：")
    member["name_kanji"] = input("名前を漢字で入力してください(例：山田　太郎)：")
    member["name_kana"] = input("名前をカタカナで入力してください(例：ヤマダ　タロウ)：")
    member["sex_division"] = input("性別を入力してください(男性/女性/その他)：")
    member["birth_day"] = input("誕生日を入力してください(例：1990/01/01)：")
    member["address"] = input("住所を入力してください：")

    print(member)
    members[7] = member
    print(members)


# ３．会員情報検索
def member_search():
    global members
    member_id_select = int(input("表示したいユーザーのIDを選んでください："))
    print(members[member_id_select])


# ４．会員情報変更
def edit():
    global member
    print("1.メールアドレス：%s" %(member["mail_address"]))
    print("2.電話番号：%s" %(member["phone_number"]))
    print("3.名前(漢字)：%s" %(member["name_kanji"]))
    print("4.名前(カナ)：%s" %(member["name_kana"]))
    print("5.性別：%s" %(member["sex_division"]))
    print("6.生年月日：%s" %(member["birth_day"]))
    print("7.住所：%s" %(member["address"]))

    options_select = int(input("何を変更しますか？："))
    member_edit = input("何に変更しますか：")

    options={
        1 : "mail_address",
        2 : "phone_number",
        3 : "name_kanji",
        4 : "name_kana",
        5 : "sex_division",
        6 : "birth_day",
        7 : "address"
    }

    member[options.get(options_select)] = member_edit

    print("変更が完了しました")
    print(member)

File: MinoriOrita/module_oriented/member_module.rb
#会員管理
module MemberModule
#認証モジュールを読み込む
require('./authentication_module.rb')
  #会員登録
  def member_register(members)
    p "mail_address:"

    mail_address = gets.chomp
    p "phone_number:"
    phone_number = gets.chomp
    p "name_kanji:"
    name_kanji = gets.chomp
    p "name_kana:"
    name_kana = gets.chomp
    p "sex_division:"
    sex_division = gets.chomp
    p "birth_day:"
    birth_day = gets.chomp
    p "address"
    address = gets.chomp
    members[members.length] = {
      'mail_address' => mail_address,
      'phone_number' => phone_number,
      'name_kanji' => name_kanji,
      'name_kana' => name_kana,
      'sex_division' => sex_division,
      'birth_day' => birth_day,
      'address' => address
    }
    p members[members.length-1]
  end

  #会員検索
  def member_search(login_status,members)
    login_status = $login_status
    if $login_status == true
      p "put member ID"
      p "member ID:"
      member_id = gets.chomp.to_i
      p members[member_id]
    else
      p "please login"
      AuthenticaionModule.login_logout(login_status)
      if $login_status == true
        p "you can search a member"
        MemberModule.member_search(login_status,members)
      end
    end
  end

  #会員編集
  def member_edit(login_status,members,item)
    login_status = $login_status
    if $login_status == true
    p "member ID:"
    member_id = gets.chomp.to_i
    p members[member_id]
    p "---------------------"
    p "0.mail_address"
    p "1.phone_number"
    p "2.name_kanji"
    p "3.sex_division"
    p "4.birth_day"
    p "5.address"
    p "---------------------"
    p "select item number"
    item_number = gets.chomp.to_i
    item_name = item[item_number]
    p members[member_id][item_name]
    p "input new"+" "+ "#{item_name}" + ":"
    members[member_id][item_name] = gets.chomp
    p members[member_id][item_name]
  else
    p "please login"
    AuthenticaionModule.login_logout(login_status)
    if $login_status == true
      p "you can edit information"
      MemberModule.member_edit(login_status,members,item)
    end
   end
  end

  module_function :member_register,:member_search,:member_edit
end
